torch/onnx/_globals.py

```diff
@@ -24,10 +24,24 @@ class _InternalGlobals:
 
     def __init__(self):
         self._export_onnx_opset_version = _constants.onnx_default_opset
+        self._training_mode: Optional[_C_onnx.TrainingMode] = None
```

Comment:
How about setting default to EVAL and drop the `Optional` (here and at the context manager)?

Reply:
Done